modules/services/dwm-status.nix

```diff
@@ -0,0 +1,70 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.dwm-status;
+
+  features = [ "audio" "backlight" "battery" "cpu_load" "network" "time" ];
+
+  configText = builtins.toJSON ({ inherit (cfg) order; } // cfg.extraConfig);
+
+  configFile = pkgs.writeText "dwm-status.json" configText;
+in
+
+{
+  options = {
+    services.dwm-status = {
+      enable = mkEnableOption "dwm-status user service";
+
+      package = mkOption {
```

Comment:
Was just about to merge this but started thinking about the `enableAlsaUtils` and `enableNetwork` parameters. Would it not be possible to have these be actual options (of `bool` type) in this module? One could then have
```
ExecStart =
  let
    package = pkgs.dwm-status.override {
      inherit (cfg) enableAlsaUtils enableNetwork;
    }
  in
    "${package}/bin/dwm-status ${configFile}";
```

and simply remove the `services.dwm-status.package` option.

Reply:
I had a similar discussion in my PR in nixpkgs. These two package overrides are for users who want minimal closure size. If you disable one of these options, you need to recompile the complete rust binary, which is unreasonable for the average user. Without providing these as separate option, the temptation is not that big to use these options.

Reply:
Ok, fair enough. I'll merge as-is 🙂

But if this is only for extreme cases then I would suggest simply removing this option and letting sufficiently interested users override the dwm-status attribute using an overlay or `nixpkgs.config.packageOverrides`.

Reply:
I tried to be consistent with most modules in nixpkgs with the `package` option.